src/main/java/com/mixpanel/android/viewcrawler/ViewSnapshot.java

```diff
@@ -232,6 +234,21 @@ private void addProperties(JsonWriter j, View v)
         }
     }
 
+    // we only do layouts for views with RelativeLayout for now
+    private void addLayouts(JsonWriter j, View v)
+        throws IOException {
+        if (v.getParent() instanceof RelativeLayout) {
```

Comment:
this is needed because we do (RelativeLayout.LayoutParams) (v.getLayoutParams()) below. and if the parent is not relativelayout, this conversion will fail. also, if we don't do this conversion here, we can't call getRules on the params. so, I'd prefer to do this check here, it makes everything much easier.